ghtt/assignment.py

```diff
@@ -627,8 +630,19 @@ def remove_grant(ctx, yes, students=None, groups=None):
                     invitation.invitee, repo.name), fg="green")
                 invitation.delete()
                 g_repo.remove_invitation(invitation.invite_id)
-        # Remove user from collaborators
-        for username in [s.username for s in repo.students]:
-            click.secho("Removing '{}' as collaborators from '{}'".format(
-                username, repo.name), fg="green")
-            g_repo.remove_from_collaborators(username)
+
+        if not read_only:
+            # Remove user from collaborators
+            for username in [s.username for s in repo.students]:
+                click.secho("Removing '{}' as collaborators from '{}'".format(
+                    username, repo.name), fg="green")
+                g_repo.remove_from_collaborators(username)
+        else:
+            # Set Read-only repo access
+            for username in [s.username for s in repo.students]:
+                current_permission = g_repo.get_collaborator_permission(username)
+                click.secho("'{}' currently has '{}' access".format(username, current_permission))
+                if current_permission == 'read':
+                    continue
+                click.secho("Setting repo access to read-only for '{}' in '{}'".format(username, repo.name), fg="green")
+                g_repo.add_to_collaborators(username, 'pull')
```

Comment:
You don't have to check if the student is already read-only, since the `add_to_collaborators` call is idempotent.

```suggestion
                click.secho("Setting repo access to read-only for '{}' in '{}'".format(username, repo.name), fg="green")
                g_repo.add_to_collaborators(username, 'pull')
```

Reply:
This was needed just for the feedback to the ghtt user. But it's not really important to know if anything changed or not. So making it simpler is OK.